docs/tutorial.md

```diff
@@ -0,0 +1,312 @@
+                container_type=ContainerType.well_plate,
+                plate_well_count=TQuantityValueNumber(value=data.number_of_wells),
+                device_system_document=DeviceSystemDocument(
+                    model_number=data.instrument.serial_number,
+                    device_identifier=data.instrument.nickname,
+                ),
+                measurement_document=self._get_measurement_document(data),
+            )
+        )
+```
+
+The first part of this method looks at the `Data` object
+defined in `example_weyland_yutani_structure.py`
+to make sure that the plate actually has some wells,
+and if not,
+raises an `AllotropeConversionError` exception.
+Please use this exception for *all* parsing and data conversion errors
+rather than (for example) raw Python `assert` statements
+so that client applications can catch problems reliably.
+
+The second part of `_get_model` constructs an instance of
+the `allotropy` class `Model`,
+which for our data is just a wrapper around another `allotropy` class
+called `MeasurementAggregateDocument`.
+Again,
+this extra level of wrapping wouldn't be necessary
+if we were only supporting one kind of machine with one kind of output,
+but a `Model` can contain other information to support other kinds of hardware.
+The various objects that `MeasurementAggregateDocument` depends on,
+such as `ContainerType` and `DeviceSystemDocument`,
+are also defined by `allotropy`;
+in practice,
+the easiest way to navigate them
+is to copy and modify an existing parser.
+
+## Making the Parser Available
+
+Once our classes and parser are defined,
+we add entries to `src/allotropy/parser_factory.py`
+to make them available to users of the package:
+
+1.  Import `ExampleWeylandYutaniParser`.
+    We don't need to import the dataclasses it depends on,
+    since client code will always access these by field name.
+
+2.  Add an entry `EXAMPLE_WEYLAND_YUTANI` to the `Vendor` enumeration
+    to uniquely identify this kind of parser.
+
+3.  Add a similar entry to the `_VENDOR_TO_PARSER` lookup table
+    to translate this enumeration element into the appropriate class.
+
+## Testing
+
+Tests for this new code go in `./tests/parsers/example_weyland_yutani/`.
+To write these,
+we create two input CSV files,
+which we put in the `./testdata` directory below our tests directory.
+We then use the command-line program `./scripts/csv_as_json.py`
+to convert these CSV files to JSON,
+inspect the JSON to make sure it's correct,
+and save it in the same `./testdata` directory.
+Our tests then compare the generated JSON against the saved files.
+More complex parsers may require tests against helper methods,
+and will probably have many more input-output pairs to check.
+
+## Exercises
+
+To check your understanding,
+look for the keyword `TODO` in the Weyland-Yutani example's source code
+and fill in the missing methods.
+You can submit your work as a pull request for review,
```

Comment:
were you planning to look at any pull requests that came in? I wouldn't want to place that burden on you, but I don't anticipate that we at Benchling would plan to look at these. as such, I would maybe frame the `TODOs` as more like thought exercises or extensions? something like:

> We've left the keyword `TODO` throughout the Weyland-Yutani example's source code in a handful of missing methods. These serve as thought exercises and examples of potential extensions to the base parser when handling more-complex files.

Reply:
updated - thank you